lib/fsm/logger.go

```diff
@@ -38,35 +40,68 @@ type Logger struct {
 	Operator ops.Operator
 	// Server is the optional server that will be attached to log entries
 	Server *storage.Server
+
+	// logEntryC is used to queue log entries and unblock execution while etcd is down during upgrades
+	// this has the potential to lose queued log messages if the process dies while etcd is down
+	logEntryC chan ops.LogEntry
+
+	// logEntryOnce bootstraps the LogEntry queue on the first log entry
+	logEntryOnce sync.Once
+}
+
+func (l *Logger) initQueue() {
+	l.logEntryOnce.Do(func() {
+		// initialize the queue to a reasonably large value, to queue all the messages during etcd upgrade
+		l.logEntryC = make(chan ops.LogEntry, 4096)
+		go l.runQueue()
+	})
+}
+
+func (l *Logger) runQueue() {
```

Comment:
I'm not sure we want to go through this type of investment for such a minor component. For now I've just changed it so that it spins up a single global queue. I see no reason that a single queue would need to support exit, and can just live for the lifetime of the process.